modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMStructureLocalServiceTest.java

```diff
@@ -892,6 +1002,33 @@ public void testSearchGlobalSiteStructure() throws Exception {
 		_userLocalService.deleteUser(user);
 	}
 
+	@Test
+	public void testUpdateDDMStructure() throws Exception {
+		DDMStructure structure = _addStructure(RandomTestUtil.randomString());
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		structure.setExternalReferenceCode(externalReferenceCode);
+
+		structure = updateStructure(structure);
+
+		Assert.assertEquals(
+			externalReferenceCode, structure.getExternalReferenceCode());
+	}
+
+	@Test(expected = DuplicateDDMStructureExternalReferenceCodeException.class)
+	public void testUpdateDDMStructureWithExistingExternalReferenceCode()
```

Comment:
You should cover this scenario inside the method `testUpdateDDMStructure()` using `AssertUtils.assertFailure`.